cmd/rollkit/commands/run_node.go

```diff
@@ -236,13 +245,27 @@ func addNodeFlags(cmd *cobra.Command) {
 }
 
 // startMockDAServJSONRPC starts a mock JSONRPC server
-func startMockDAServJSONRPC(ctx context.Context) (*proxy.Server, error) {
-	addr, _ := url.Parse(nodeConfig.DAAddress)
-	srv := proxy.NewServer(addr.Hostname(), addr.Port(), goDATest.NewDummyDA())
-	err := srv.Start(ctx)
+func startMockDAServJSONRPC(
+	ctx context.Context,
+	daAddress string,
+	newServer func(string, string, da.DA) *proxy.Server,
+) (*proxy.Server, error) {
+	addr, err := url.Parse(daAddress)
 	if err != nil {
 		return nil, err
 	}
+
+	srv := newServer(addr.Hostname(), addr.Port(), goDATest.NewDummyDA())
+	if err := srv.Start(ctx); err != nil {
+		if errors.Is(err, syscall.EADDRINUSE) {
+			logger.Info("Mock DA server is already running", "address", nodeConfig.DAAddress)
+			return nil, errMockDAServerAlreadyRunning
```

Comment:
It isn't necessarily the mock server that is running so this could be a confusing error message. 
```suggestion
			logger.Info("DA server is already running", "address", nodeConfig.DAAddress)
			return nil, errDAServerAlreadyRunning
```